imap_processing/hit/l1a/hit_l1a.py

```diff
@@ -50,203 +47,207 @@ def hit_l1a(packet_file: str, data_version: str) -> list[xr.Dataset]:
-    grouped_data = group_data(sorted_packets)
-
-    # create the attribute manager for this data level
+    # TODO add logging
+
+    # Unpack ccsds file
+    packet_definition = (
+        imap_module_directory / "hit/packet_definitions/" "hit_packet_definitions.xml"
+    )
+    datasets_by_apid = packet_file_to_datasets(
+        packet_file=packet_file,
+        xtce_packet_definition=packet_definition,
+    )
+
+    # Create the attribute manager for this data level
     attr_mgr = ImapCdfAttributes()
     attr_mgr.add_instrument_global_attrs(instrument="hit")
     attr_mgr.add_instrument_variable_attrs(instrument="hit", level="l1a")
     attr_mgr.add_global_attribute("Data_version", data_version)
 
-    # Create datasets
-    datasets = create_datasets(grouped_data, attr_mgr)
+    # Process science to l1a.
+    processed_data = []
+    for apid in datasets_by_apid:
+        if apid == HitAPID.HIT_HSKP:
+            housekeeping_dataset = process_housekeeping(
+                datasets_by_apid[apid], attr_mgr
+            )
+            processed_data.append(housekeeping_dataset)
+        elif apid == HitAPID.HIT_SCIENCE:
+            # TODO complete science data processing
+            print("Skipping science data for now")
+            # science_dataset = process_science(datasets_by_apid[apid], attr_mgr)
+        else:
+            raise Exception(f"Unknown APID [{apid}]")
 
-    return list(datasets.values())
+    return processed_data
 
 
-def decom_packets(packet_file: str) -> list:
+def concatenate_leak_variables(dataset: xr.Dataset) -> xr.Dataset:
     """
-    Unpack and decode packets using CCSDS file and XTCE packet definitions.
+    Concatenate leak variables in the dataset.
+
+    Updates the housekeeping dataset to replace the individual
+    leak_i_00, leak_i_01, ..., leak_i_63 variables with a single
+    leak_i variable as a 2D array. This variable represents
+    leakage current [Voltage] data.
 
     Parameters
     ----------
-    packet_file : str
-        Path to the CCSDS data packet file.
+    dataset : xarray.Dataset
+        Dataset containing 64 leak variables.
 
     Returns
     -------
-    unpacked_packets : list
-        List of all the unpacked data.
+    dataset : xarray.Dataset
+        Updated dataset with concatenated leak variables.
     """
-    # TODO: update path to use a combined packets xtce file
-    xtce_file = imap_module_directory / "hit/packet_definitions/P_HIT_HSKP.xml"
-    logger.debug(f"Unpacking {packet_file} using xtce definitions in {xtce_file}")
-    unpacked_packets: list = decom.decom_packets(packet_file, xtce_file)
-    logger.debug(f"{packet_file} unpacked")
-    return unpacked_packets
+    # Stack 64 leak variables (leak_00, leak_01, ..., leak_63)
+    leak_vars = [dataset[f"leak_i_{i:02d}"] for i in range(64)]
 
+    # Concatenate along 'leak_index' and reorder dimensions
+    stacked_leaks = xr.concat(leak_vars, dim="leak_index").transpose(
```

Comment:
should the dim here be adc_channels?

Reply:
Really nice use of the xarray functions to do this for you :)

Reply:
Correct, adc_channels is the dim for the leak variable. Dims will be updated later in the process when dims for all the fields get assigned according to the cdf yaml file definitions. I just needed to put a dimension here that made sense as a placeholder.

`    # Assign attributes and dimensions to each data array in the Dataset
    for field, data in dataset.data_vars.items():
        # Create a list of dimensions using the DEPEND_I keys in the
        # attributes
        dims = [
            value
            for key, value in attr_mgr.get_variable_attributes(field).items()
            if "DEPEND" in key
        ]
        dataset[field] = xr.DataArray(
            data,
            dims=dims,
            attrs=attr_mgr.get_variable_attributes(field),
        )

`

Reply:
I updated the function to pass in adc_channels and assign it as a dimension